feat(ws:)

Mohamed-ben-khemis opened the pull request.

ederign commented:
/ok-to-test

ederign commented on workspaces/backend/Makefile:
```diff
@@ -66,14 +66,21 @@ lint: golangci-lint ## Run golangci-lint linter
+##@ Swagger
+
+.PHONY: swag
+swag: 
```
We will have to add instructions to either user install it or make sure that we install as binary. Do you think we can do similar to what we do on bin? with env test?

ederign replied:
Also, other kubeflow projects "version" the file under api folder:
https://github.com/kubeflow/model-registry/tree/main/api/
https://github.com/kubeflow/pipelines/tree/master/api

Can we do the same? If we have the generated file inside our codebase we add easily to kubeflow docs.

ederign replied:
Similar to https://www.kubeflow.org/docs/components/model-registry/reference/rest-api/

ederign commented on workspaces/backend/api/v1/swagger/docs.go:
```diff
@@ -0,0 +1,110 @@
+// Package swagger Code generated by swaggo/swag. DO NOT EDIT
+package swagger
```
As I mention on other, my suggestion is to serialize into another folder (i.e. openapi) on root of backend

ederign reviewed (changes requested):
We are on the right path. Just added some comments there:

Also, can we reduce the logs generated?

```code
:no Go files in /Users/ederign/src/kubeflow/notebooks/workspaces/backend
2025/02/18 10:29:57 Generating health_check.HealthCheck
2025/02/18 10:29:57 Generating health_check.ServiceStatus
2025/02/18 10:29:57 Generating health_check.SystemInfo
2025/02/18 10:29:57 Generating api.ErrorResponse
2025/02/18 10:29:57 create docs.go at api/v1/swagger/docs.go
2025/02/18 10:29:57 create swagger.json at api/v1/swagger/swagger.json
2025/02/18 10:29:57 create swagger.yaml at api/v1/swagger/swagger.yaml
```

ederign commented:
Great progress @Mohamed-ben-khemis  take a look on our tests that are failing due to lack of binary.

Also, 7 warning: failed to get package name in dir: ./, error: execute go list command, exit status 1, stdout:, stderr:no Go files in /Users/ederign/src/kubeflow/notebooks/workspaces/backend

Mohamed-ben-khemis commented:
> We are on the right path. Just added some comments there:
> 
> Also, can we reduce the logs generated?
> 
> ```
> :no Go files in /Users/ederign/src/kubeflow/notebooks/workspaces/backend
> 2025/02/18 10:29:57 Generating health_check.HealthCheck
> 2025/02/18 10:29:57 Generating health_check.ServiceStatus
> 2025/02/18 10:29:57 Generating health_check.SystemInfo
> 2025/02/18 10:29:57 Generating api.ErrorResponse
> 2025/02/18 10:29:57 create docs.go at api/v1/swagger/docs.go
> 2025/02/18 10:29:57 create swagger.json at api/v1/swagger/swagger.json
> 2025/02/18 10:29:57 create swagger.yaml at api/v1/swagger/swagger.yaml
> ```
Done! I've added quiet mode to swag init
<img width="1025" alt="swag init" src="https://github.com/user-attachments/assets/07ee7a99-342d-45df-83a6-fd0fd97fc16b" />
<img width="1088" alt="Screenshot at Feb 19 14-28-27" src="https://github.com/user-attachments/assets/e512b6e2-8156-4c59-89d6-7f413a91ceb1" />

Mohamed-ben-khemis commented:
> Great progress @Mohamed-ben-khemis take a look on our tests that are failing due to lack of binary.
> 
> Also, 7 warning: failed to get package name in dir: ./, error: execute go list command, exit status 1, stdout:, stderr:no Go files in /Users/ederign/src/kubeflow/notebooks/workspaces/backend
Fixed swag init by adjusting directory selection. Now it properly processes Go files, excluding cmd/, and runs without warning.
<img width="875" alt="image" src="https://github.com/user-attachments/assets/545f0c0a-6737-4474-8f3e-0b52d0b4fdf2" />

ederign commented:
Nice work @Mohamed-ben-khemis! This is a good starting point for our swagger generation. Let's review it on the call today with @thesuperzapper, and my suggestion would be for you to start the other endpoints!

Mohamed-ben-khemis commented:
> Nice work @Mohamed-ben-khemis! This is a good starting point for our swagger generation. Let's review it on the call today with @thesuperzapper, and my suggestion would be for you to start the other endpoints!

Great! Thanks for the review @ederign. Now that this is merged, I'll begin working on the other endpoints.

thesuperzapper commented:
@Mohamed-ben-khemis thanks for you work on this, I have rebased the PR and fixed a few things so we can merge asap:

1. Fixed the error response type of healthcheck to `ErrorEnvelope`
2. Updated the `http-swagger` package to `2.0.2`
3. Factored out the swagger handler into its own file, like the other ones (see `GetSwaggerHandler`)
4. Removed the broken `swagger.yaml` http path handleing, and just directly call the `http-swagger` handler
5. Used tab-indented docstrings so gofmt does not break the formatting of the `@Summary` docstrings

Feel free to start working on documenting the other APIs in a future PR.

/lgtm
/approve
